core/src/main/java/org/opensearch/sql/analysis/AnalysisContext.java

```diff
@@ -23,13 +27,27 @@ public class AnalysisContext {
   @Getter
   private final List<NamedExpression> namedParseExpressions;
 
+  /**
+   * Storage for values of functions which return a constant value.
+   * We are storing the values there to use it in sequential calls to those functions.
+   * For example, `now` function should the same value during processing a query.
+   */
+  @Getter
+  @Setter
+  private Map<String, Expression> cachedFunctionValues;
+
   public AnalysisContext() {
```

Comment:
Is this constructor used? 

Doesn't seem to leave the object in a well-defined state since `namedParseExpressions` and cachedFunctionValues` will be null.

Reply:
Yes, https://github.com/Bit-Quill/opensearch-project-sql/blob/ce2e8bfc3c0df1ac99c1bb496459d7f781852321/sql/src/main/java/org/opensearch/sql/sql/SQLService.java#L98-L100.

These class members could be null, because they are initialized anyway in the CTOR.